src/dotnet/commands/dotnet-add-p2p/Program.cs

```diff
@@ -0,0 +1,232 @@
+            CommandOption referenceOption = app.Option("-r|--reference <REFERENCE>", "Add project to project <REFERENCE> to <PROJECT>", CommandOptionType.MultipleValue);
+
+
+            app.OnExecute(() => {
+                ProjectRootElement project = projectArgument.Value != null ?
+                                                GetProjectFromFileOrThrow(projectArgument.Value) :
+                                                GetProjectFromCurrentDirectoryOrThrow();
+
+                if (referenceOption.Values.Count == 0)
+                {
+                    throw new GracefulException("You must specify at least one reference to add.");
+                }
+
+                AddProjectToProjectReference(project, frameworkOption.Value(), referenceOption.Values);
+
+                project.Save();
+
+                return 0;
+            });
+
+            try
+            {
+                return app.Execute(args);
+            }
+            catch (GracefulException e)
+            {
+                Reporter.Error.WriteLine(e.Message);
+                return 1;
+            }
+        }
+
+        // There is ProjectRootElement.TryOpen but it does not work as expected
+        // I.e. it returns null for some valid projects
+        public static ProjectRootElement TryOpenProject(string filename)
+        {
+            try
+            {
+                return ProjectRootElement.Open(filename);
+            }
+            catch (Microsoft.Build.Exceptions.InvalidProjectFileException)
+            {
+                return null;
+            }
+        }
+
+        public static ProjectRootElement GetProjectFromFileOrThrow(string filename)
+        {
+            if (!File.Exists(filename))
+            {
+                throw new GracefulException($"Provided project `{filename}` does not exist.");
+            }
+
+            var project = TryOpenProject(filename);
+            if (project == null)
+            {
+                throw new GracefulException($"Invalid MSBuild project `{filename}`.");
+            }
+
+            return project;
+        }
+
+        public static ProjectRootElement GetProjectFromCurrentDirectoryOrThrow()
+        {
+            DirectoryInfo currDir = new DirectoryInfo(Directory.GetCurrentDirectory());
+            FileInfo[] files = currDir.GetFiles("*proj");
+            if (files.Length == 0)
+            {
+                throw new GracefulException("Could not find any MSBuild project in the current directory.");
+            }
+
+            if (files.Length > 1)
+            {
+                throw new GracefulException("Found more than one MSBuild project in the current directory. Please specify which one to use.");
+            }
+
+            FileInfo projectFile = files.First();
+
+            if (!projectFile.Exists)
```

Comment:
Based on our conversation in standup:
- dotnet-add will:
  - determine if it was passed a path
    - if YES it will fully qualify the path and ensure it exists, providing associated error messages
    - if NO it will identify the `pwd`
- dotnet-add-{specialization} will:
  - if args[0] is a directory then it scans the directory for an appropriate Object
    - if 0 || multiple objects, it returns a meaningful error
    - else it has a full path to the Object
  - Now it has a full path to the Object, one way or another, and can proceed to the modification implementation

Thoughts?